openwisp_notifications/tokens.py

```diff
@@ -0,0 +1,68 @@
+from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.utils.crypto import constant_time_compare
+from django.utils.http import base36_to_int
+
+
+class EmailTokenGenerator(PasswordResetTokenGenerator):
+    """
+    Email token generator that extends the default PasswordResetTokenGenerator
+    with a fixed 7-day expiry period and a salt key.
+    """
+
+    key_salt = "openwisp_notifications.tokens.EmailTokenGenerator"
+
+    def __init__(self):
+        super().__init__()
+        self.expiry_days = 7
```

Comment:
@Dhanus3133 @pandafy did we talk about setting an expiration to the date? I think we said to not have any time based expiration?

Reply:
I have now removed it.